src/prng.rs

```diff
@@ -76,23 +74,51 @@ where
             // Seek to the next chunk of the buffer that encodes an element of F.
             for i in (self.buffer_index..self.buffer.len()).step_by(F::ENCODED_SIZE) {
                 let j = i + F::ENCODED_SIZE;
+
+                if j > self.buffer.len() {
+                    break;
+                }
+
+                self.buffer_index = j;
+
                 if let Some(x) = match F::try_from_random(&self.buffer[i..j]) {
                     Ok(x) => Some(x),
                     Err(FieldError::ModulusOverflow) => None, // reject this sample
                     Err(err) => panic!("unexpected error: {err}"),
                 } {
-                    // Set the buffer index to the next chunk.
-                    self.buffer_index = j;
-                    self.output_written += 1;
                     return x;
                 }
             }
 
-            // Refresh buffer with the next chunk of PRG output.
-            self.seed_stream.fill(&mut self.buffer);
+            // Refresh buffer with the next chunk of PRG output, filling the front of the buffer
+            // with the leftovers. This ensures continuity of the seeds stream after converting the
+            // `Prng` to a new field type via `into_new_field()`.
+            let left_over = self.buffer.len() - self.buffer_index;
+            for i in 0..left_over {
+                self.buffer[i] = self.buffer[self.buffer_index + i];
+            }
+            self.seed_stream.fill(&mut self.buffer[left_over..]);
             self.buffer_index = 0;
         }
     }
+
+    /// Convert this object into a field element generator for a different field.
+    //
+    // TODO(cjpatton) spec: Consider using distinct seeds for distinct field types. Buffering the
```

Comment:
If we don't make this spec change, we should have some additional PRG test vectors to cover generating elements in multiple fields from one hash input.

Reply:
Good idea, I'll make a note. The `left_over_buffer_back_fill` test is meant to replicate the conditions that lead to this, but it's sensible to just have test vectors as well.

I'm writing up an issue for the VDAF spec summarizing all of the changes I'd like to make.